src/sentry/static/sentry/app/views/onboarding/progress.jsx

```diff
@@ -36,10 +38,39 @@ const ProgressNodes = createReactClass({
 
   steps: Object.keys(onboardingSteps),
 
+  getSteps() {
+    let organization = this.context.organization;
+
+    return (
+      HookStore.get('component:onboarding-sidebar').length &&
+      HookStore.get('component:onboarding-sidebar')[0]('steps', {
+        onboardingSteps,
+        organization,
+      })
+    );
+  },
+
+  getStepDescriptions() {
+    let organization = this.context.organization;
+
+    return (
+      HookStore.get('component:onboarding-sidebar').length &&
+      HookStore.get('component:onboarding-sidebar')[0]('stepDescriptions', {
+        stepDescriptions,
+        organization,
+      })
+    );
+  },
+
   inferStep() {
-    let {projectId} = this.props.params;
-    if (!projectId) return onboardingSteps.project;
-    return onboardingSteps.configure;
+    let {pathname} = this.context.location;
+    let {params} = this.props;
+    let steps = this.getSteps();
+
+    if (!params.projectId) return steps.project;
+    if (pathname.indexOf('/survey/') !== -1) return steps.survey;
```

Comment:
suppose we move the url into getsentry, would it be appropriate for sentry to know about the url then?